sound/soc/sof/intel/telemetry.h

```diff
@@ -0,0 +1,223 @@
+	uint32_t max_sys_tick_count;
+	uint32_t last_ccount;
+	uint32_t avg_utilization;
+	uint32_t peak_utilization;
+	uint32_t peak_utilization_4k;
+	uint32_t peak_utilization_8k;
+	uint32_t rsvd[2];
+};
+
+struct deadlock_info_s {
+	uint32_t register_a0;
+	uint32_t register_a1;
+	uint32_t cached_stack_ptr;
+	uint32_t run_condition_data[9];
+};
+
+struct assert_statistics_s {
+	uint32_t asserts_triggered_count;
+	uint32_t last_assert_log_id;
+};
+
+struct assert_info_s {
+	uint32_t            mode;
+	struct assert_statistics_s stats[SOF_IPC4_FW_REPORTED_MAX_CORES_COUNT];
+};
+
+struct xxxruns_info_s {
+	uint32_t gateway_underruns_count;
+	uint32_t last_underrun_gateway_node_id;
+	uint32_t gateway_overruns_count;
+	uint32_t last_overrun_gateway_node_id;
+	uint32_t mixer_underruns_count;
+	uint32_t last_mixer_underrun_resource_id;
+};
+
+struct performance_info_s {
+	uint32_t cpu_frequency;
+	uint32_t dropped_lp_timer_ticks_cause[SOF_IPC4_FW_LP_TIMER_DELAY_CAUSE_COUNT];
+};
+
+struct sram_mem_info_s {
+	uint32_t total_pages_cnt;
+	uint32_t pages_in_use_cnt;
+    uint32_t phys_pages_in_use_cnt;
+};
+
+struct sram_mem_pool_info_s {
+	uint32_t registered_pools_cnt;
+	uint32_t pools_in_use_cnt;
+	uint32_t registered_pages_cnt;
+	uint32_t pages_in_use_cnt;
+};
+
+struct sram_oma_mem_pool_info_s
+{
+    uint32_t registered_oma_pools_cnt;
+    uint32_t oma_pools_in_use_cnt;
+    uint32_t registered_oma_pages_cnt;
+    uint32_t oma_pages_in_use_cnt;
+    uint32_t requested_oma_pages_cnt;
+};
+
+struct mem_pools_info_s {
+    struct sram_mem_info_s hpsram_mem_info;
+    struct sram_mem_pool_info_s hpsram_mem_pool_info;
+    struct sram_oma_mem_pool_info_s hpsram_oma_pool_info;
+    struct sram_mem_info_s lpsram_mem_info;
+    struct sram_mem_pool_info_s lpsram_mem_pool_info;
+    struct sram_oma_mem_pool_info_s lpsram_oma_pool_info;
+};
+
+struct timeout_info_s {
+	uint32_t expected_value;
+	uint32_t current_value;
+};
+
+struct ulp_prevention_info_s {
+	uint8_t preventedby_high_power;
+	uint8_t preventedby_idle_mode_count_low;
+	uint8_t idle_mode_count_down_;
+	uint8_t preventedby_on_pipeline_state;
+	uint8_t preventedby_fastmode_task;
+	uint8_t prevent_autonomus_reset_counter;
+    uint8_t rsvd[2];
+};
+
+struct transition_info_s {
+	uint32_t entry_cnt;
+	uint32_t exit_cnt;
+	uint32_t last_entry_systick;
+};
+
+struct vad_kpd_stats_s {
+    uint32_t vad_cnt;
+    uint32_t aad_cnt;
+    uint32_t nld_cnt;
+    uint32_t kpd_cnt;
+};
+
+struct ulp_telemetry_s {
+	struct ulp_prevention_info_s prevention_info;
+	struct transition_info_s     transition_info;
+	struct vad_kpd_stats_s       vad_kpd_stats;
+};
+
+struct thread_entry_point_info_s {
+	uint32_t call_stack;
+	uint32_t start_systick;
+};
+
+struct task_info_s {
+	uint32_t task_id;
+	uint8_t task_state;
+	uint8_t task_type;
+	uint8_t task_subtype;
+	uint8_t core_id;
+	struct thread_entry_point_info_s blockade_entry_point;
+	struct thread_entry_point_info_s cs_entry_point;
+};
+
+struct CoreExceptionRecord {
```

Comment:
core_exception_record? Why the CamelCaseHere?

Reply:
thanks, I will refine it.